core/services/functions/orm_test.go

```diff
@@ -74,8 +75,12 @@ func TestORM_CreateRequestsAndFindByID(t *testing.T) {
 	t.Run("duplicated", func(t *testing.T) {
 		err := orm.CreateRequest(id1, ts1, &txHash1)
 		require.Error(t, err)
+		require.True(t, errors.Is(err, functions.ErrDuplicateRequestID))
 		err = orm.CreateRequest(id1, ts1, &txHash1)
 		require.Error(t, err)
+		// QQQ: the first error type check above passes but his one
+		// returns a "current transaction is aborted" - why?
+		require.True(t, errors.Is(err, functions.ErrDuplicateRequestID))
 	})
```

Comment:
Not sure off the top of my head, but one thing you could improve about this is instead of `require.True` use:
`require.ErorrIs(t, err, functions.ErrDuplicateRequestID)`

That will print the actual and expected errors so you can compare, instead of just "Should be true" if it fails.

Reply:
Oh I see what the problem is... you're looking for 2 sql errors in a row to happen, but that can never happen in tests.  The tests run in a special mode where everything is wrapped inside a single transaction.  So you can never get more than one error... all queries after the first error will fail with error message "transaction is aborted".